cli/src/test-utils/test-runner.ts

```diff
@@ -0,0 +1,300 @@
+import JsonSchemaValidator from "ajv";
+import axios, { AxiosRequestConfig } from "axios";
+import {
+  JsonSchemaType,
+  jsonTypeSchema
+} from "../../../lib/src/generators/contract/json-schema";
+import {
+  ContractDefinition,
+  DefaultResponseDefinition,
+  EndpointDefinition,
+  TestDefinition
+} from "../../../lib/src/models/definitions";
+import { TypeNode } from "../../../lib/src/models/nodes";
+import { DataType } from "../../../lib/src/models/types";
+import { valueFromDataExpression } from "../../../lib/src/utilities/data-expression-utils";
+import { asyncForEach } from "../common/utils";
+import TestLogger from "./test-logger";
+
+export async function runTest(
+  definition: ContractDefinition,
+  stateUrl: string,
+  baseUrl: string,
+  testFilter?: TestFilter
+): Promise<boolean> {
+  let allPassed = true;
+
+  await asyncForEach(definition.endpoints, async endpoint => {
```

Comment:
Why not use a simple for loop?

Reply:
`forEach` syntax offers improved readability. This is even especially important in this case where we have a nested loop - cognitive overhead of keeping track of multiple indexes